### ISSUE TITLE
Cannot put spaces in project names - Maui then attempts to use underscores in App manifest name which are not allowed - causes failure to build

### ISSUE THREAD

**Issue by jonmdev** [dotnet/maui#21172]
### Description

This is something I reported here: https://github.com/dotnet/maui/issues/18177

Then it was marked as fixed. It is still a problem.

If you click File > New and select Maui Project and then name your project something with spaces (eg. "Project Name Spaces Bug") in current .NET 8.0, attempts to build will return:

```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error		DEP0700: Registration of the app failed. [0x80080204] error 0xC00CE169: App manifest validation error: The app manifest must be valid as per schema: Line 10, Column 13, Reason: 'com.companyname.project_name_spaces_bug' violates pattern constraint of '[-.A-Za-z0-9]+'.
The attribute 'Name' with value 'com.companyname.project_name_spaces_bug' failed to parse.	Project Name Spaces Bug				
```

This is in version: 8.0.100 which is what Visual Studio is using by default.

If we are not supposed to be using this version, what are we supposed to be using and how do we use it? It is simply loading this by default.

The bug as noted in the previous report is that the "pattern constraint" should allow underscores in the name (as it used to) but is not.

### Steps to Reproduce

1. File > New
2. Name it something with spaces
3. Try to build/debug

### Link to public reproduction project repository

https://github.com/jonmdev/Project-Name-Spaces-Bug

### Version with bug

8.0.3 GA

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.101

### Affected platforms

iOS, Android, Windows, macOS

### Did you find any workaround?

Don't use spaces in any project names?

**Comment by jaosnz-rep** [comment]
Verified on 17.10.0 Preview 2.0, can repro on Windows platform with sample project.
https://github.com/jonmdev/Project-Name-Spaces-Bug

**Comment by mattleibow** [comment]
This was fixed in 8.0.7: https://github.com/dotnet/maui/commit/e9029d0a177a8b99d720ce29bd2afd2ef644c92a
